nav2_docking/opennav_docking/src/controller.cpp

```diff
@@ -39,6 +49,22 @@ Controller::Controller(const rclcpp_lifecycle::LifecycleNode::SharedPtr & node)
     node, "controller.v_angular_max", rclcpp::ParameterValue(0.75));
   nav2_util::declare_parameter_if_not_declared(
     node, "controller.slowdown_radius", rclcpp::ParameterValue(0.25));
+  nav2_util::declare_parameter_if_not_declared(
+    node, "controller.use_collision_detection", rclcpp::ParameterValue(true));
+  nav2_util::declare_parameter_if_not_declared(
```

Comment:
Need to add this to the migration guide + add the new parameters to the configuration guide + add new parameters to the `nav2_params.yaml` for example. The `collision_tolerance` especially should be described with some care about what it represents. I think this should _probably_ be renamed to something like `dock_collision_checking_tolerance` or `dock_collision_distance` or something more clear